### PR TITLE
Automated vanilla image creation via VNC

### PR DESCRIPTION
**Pull request by edigaryev**
Requires https://github.com/cirruslabs/packer-plugin-tart/pull/21.

Needs `0.5.0` release of https://github.com/cirruslabs/packer-plugin-tart.

### CODE REVIEW

**Review by fkorotkov — COMMENTED** [review]
I think it's worth updating ReadMe as well to highlight there is no need for manual steps anymore.

File: templates/vanilla-monterey.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+}
+
+source "tart-cli" "tart" {
+  from_ipsw    = "latest"
+  vm_name      = "monterey-vanilla"
+  cpu_count    = 4
+  memory_gb    = 8
+  disk_size_gb = 50
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply templates/vanilla-monterey.pkr.hcl:15]: > Why so large?

It's the default disk size that Tart creates: https://github.com/cirruslabs/tart/blob/0229138bd5947410c3c0ef8dfac3cf57e537cf78/Sources/tart/Commands/Create.swift#L19.

[the remark this answers is not in the document]
Comment (fkorotkov) [reply templates/vanilla-monterey.pkr.hcl:15]: I think we can do something like https://github.com/cirruslabs/packer-plugin-tart/pull/23 to respect `disk_size_gb` upon creation.

File: templates/vanilla-ventura.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  required_plugins {
+    tart = {
+      version = ">= 0.5.1"
+      source  = "github.com/cirruslabs/tart"
+    }
+  }
+}
+
+source "tart-cli" "tart" {
+  from_ipsw    = "https://updates.cdn-apple.com/2022SummerSeed/fullrestores/012-70113/6F1F08B7-9A1B-48A9-93DB-55EE21121C87/UniversalMac_13.0_22A5352e_Restore.ipsw"
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply templates/vanilla-ventura.pkr.hcl:13]: See https://github.com/cirruslabs/macos-image-templates/pull/55/commits/57792c0ba0ed36284d0900a92e55dc95d8ccf3e8.

File: templates/vanilla-ventura.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    "<wait10s><leftCtrlOn><f2><leftCtrlOff><right><right><right><right><down><down>sharing<enter>",
+    # Navigate to "Remote Login" and enable it
+    "<wait10s><tab><tab><tab><tab><tab><tab><tab><tab><spacebar>",
+    # Navigate to "Remote Login" once more and open its settings
+    "<wait10s><tab><tab><tab><tab><tab><tab><tab><tab><spacebar>",
+    # Enable "Full Disk Access"
+    "<wait10s><tab><spacebar>",
+    # Click "Done"
+    "<wait10s><leftShiftOn><tab><leftShiftOff><leftShiftOn><tab><leftShiftOff><spacebar>",
+    # Disable Voice Over
```
[the remark this answers is not in the document]
Comment (raven) [reply templates/vanilla-ventura.pkr.hcl:74]: This is what I have for Ventura System Settings dance:

```
    
    # --- Remote login with full disk access
    # open "System Settings"
    "<wait5s><leftAltOn><spacebar><leftAltOff>System Settings<enter>",
    # Navigate to "Sharing"
    "<wait3s><leftAltOn>f<leftAltOff>sharing<return>",
    # Navigate to "Remote Login" and enable it
    "<wait3s><tab><tab><tab><tab><tab><tab><tab><tab><tab><tab><spacebar>",
    # Enable "Full Disk Access" for remote login
    "<wait3s><tab><tab><tab><tab><tab><tab><tab><tab><tab><tab><spacebar>",
    "<wait3s><tab><spacebar><leftShiftOn><tab><tab><leftShiftOff><spacebar>",
    # Close System Settings
    "<wait3s><leftAltOn>w<leftAltOff>",

    # --- Disable screen saver and lock screen
    # open "System Settings"
    "<wait5s><leftAltOn><spacebar><leftAltOff>System Settings<enter>",
    # Navigate to "lock screen"
    "<wait3s><leftAltOn>f<leftAltOff>lock screen<return>",
    # Select 'Never' for Start Screen Saver when inactive
    "<wait3s><tab><spacebar>Never<return>",
    # Select 'Never' and enter password for Require password after screen saver begins 
    "<wait3s><tab><tab><spacebar>Never<return>",
    "<wait3s>admin<return>",
    "<wait3s><spacebar>",
    # Close System Settings
    "<wait3s><leftAltOn>w<leftAltOff>",

    # --- Automatic login
    # open "System Settings"
    "<wait5s><leftAltOn><spacebar><leftAltOff>System Settings<enter>",
    # Navigate to "Users & Groups"
    "<wait3s><leftAltOn>f<leftAltOff>users groups<return>",
    # Select 'admin' and enter password for two dialogues for Automatically log in as...
    "<wait3s><tab><tab><tab><tab><tab><spacebar>",
    "<wait3s>admin<return>",
    "<wait3s>admin<return>",
    "<wait3s>admin<return>",
   # Close System Settings
    "<wait3s><leftAltOn>w<leftAltOff>",
```

[the remark this answers is not in the document]
Comment (raven) [reply templates/vanilla-ventura.pkr.hcl:74]: Adding admin to sudoers:  
There's probably a better way, as we have already enabled `ssh` access, but if doing it via VNC key commands 🤮 
```
    # --- Adding admin to sudoers
    "<wait5s><leftAltOn><spacebar><leftAltOff>terminal<return>",
    "<wait3s>sudo visudo<return>",
    "<wait3s>admin<return>",
    "<wait3s><leftShiftOn>g<leftShiftOff>o",
    "<wait3s>admin<spacebar><leftShiftOn>all<leftShiftOff><spacebar>=<spacebar><leftShiftOn>9all0<spacebar>nopasswd;all<leftShiftOff><esc><leftShiftOn>;<leftShiftOff>wq<return>",
    "<wait3s>",
    ```

[the remark this answers is not in the document]
Comment (edigaryev) [reply templates/vanilla-ventura.pkr.hcl:74]: >Is enabling automatic login no longer required?

It is, it's just still WIP.

>There's probably a better way, as we have already enabled ssh access, but if doing it via VNC key commands 🤮

You're right — no need for VNC for this one. Take a look at https://github.com/cirruslabs/macos-image-templates/pull/55/commits/e9d019600e71b95a044f218669220d6369569fee.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**